src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -94,13 +99,13 @@ private void SetCursor(Cursor cursor, StreamSequenceToken sequenceToken)
                 cursor.State = CursorStates.Idle;
                 cursor.CurrentBlock = newestBlock;
                 cursor.Index = newestBlock.Value.NewestMessageIndex;
-                cursor.SequenceToken = newestBlock.Value.NewestSequenceToken;
+                cursor.SequenceToken = newestBlock.Value.GetNewestSequenceToken(cacheDataAdapter);
                 return;
             }
 
             // If sequenceToken is too new to be in cache, unset token, and wait for more data.
             TCachedMessage newestMessage = newestBlock.Value.NewestMessage;
-            if (cacheDataAdapter.CompareCachedMessageToSequenceToken(ref newestMessage, sequenceToken) < 0) 
+            if (comparer.Compare(newestMessage, sequenceToken) < 0) 
```

Comment:
The ref was for perf reasons.  Passing structures by ref is faster than by value because they are reference types.  I won't know the effect of this until I do performance testing.  I removed it because I concluded it was a premature micro-optimization.